File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        n = x
        r = 0
        while n > 0:
            d = n % 10
            r = r * 10
            r = r + d
            n = n // 10
        
        if r == x:
            return True
        return False   

File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0) return false;
        if(x>-1 && x<10) return true;
        vector<int>ans;
        int digit = 0;
        while(x>0){
            digit = x%10;
            ans.push_back(digit);
            x = x/10;
        }
        int n = ans.size();
        for(int i=0;i<ans.size()/2;i++){
            if(ans[i]!=ans[n-1-i]){
                return false;
            }
        }
        return true;
    }
};